### DISCUSSION TITLE
Doveadm - Error: auth-master: userdb lookup

### DISCUSSION THREAD

**Discussion by jularnaud** [docker-mailserver/docker-mailserver#2950] [General]
Hello there,

apparently there is a [bug](https://www.mail-archive.com/dovecot@dovecot.org/msg86160.html) on Dovecot 2.3.13 -> 2.3.19 concerning the search of multiple userdb :

doveadm(email0@email0.com)<5540><>: Error: auth-master: userdb lookup(email0@email0.com): connect(/run/dovecot/auth-userdb) failed: No such file or directory
doveadm(email0@email0.com): Error: User lookup failed: Internal error occurred. Refer to server log for more information.
[  ERROR  ]  Supplied non-number argument '' to '_bytes_to_human_readable_size()'
[  ERROR  ]  Aborting
[  ERROR  ]  Supplied non-number argument '' to '_bytes_to_human_readable_size()'
[  ERROR  ]  Aborting
* email0@email0.com (  /  ) [%]

doveadm(email1@email1.com)<5540><>: Error: auth-master: userdb lookup(email1@email1.com): connect(/run/dovecot/auth-userdb) failed: No such file or directory
doveadm(email1@email1.com): Error: User lookup failed: Internal error occurred. Refer to server log for more information.
[  ERROR  ]  Supplied non-number argument '' to '_bytes_to_human_readable_size()'
[  ERROR  ]  Aborting
[  ERROR  ]  Supplied non-number argument '' to '_bytes_to_human_readable_size()'
[  ERROR  ]  Aborting
* email1@email1.com (  /  ) [%]

Is there any plans to upgrade Dovecot with 2.3.20?

Thanks,

**Comment by casperklein** [comment]
At the moment, Dovecot (2.3.13) from the Debian repository is installed. However, in the next release we'll switch to Dovecots own repository, which got the latest released version.

For now, you can build the image yourself, with `DOVECOT_COMMUNITY_REPO=1` to get the latest Dovecot version:

`DOCKER_BUILDKIT=1 docker build --build-arg DOVECOT_COMMUNITY_REPO=1 -t MY-DMS .`

**Comment by jularnaud** [comment]
Thanks for your reply,
when i try to built the image, i got this error :
=> CANCELED [ 2/46] RUN   apt-get -qq update &&   apt-get -qq install apt-utils 2>/dev/null &&   apt-get -qq dist-upgrade &&   echo "applying workaround for ubuntu/postfix bug described in https://github.com/docker-mailserver/docker-mailserver/issues/2023#issuec  1.3s
 
=> ERROR [43/46] COPY   ./target/bin/*   ./target/scripts/*.sh   ./target/scripts/startup/*.sh   ./target/scripts/wrapper/*.sh   ./target/docker-configomat/configomat.sh   /usr/local/bin/                                                                             0.0s
------
 > [43/46] COPY   ./target/bin/*   ./target/scripts/*.sh   ./target/scripts/startup/*.sh   ./target/scripts/wrapper/*.sh   ./target/docker-configomat/configomat.sh   /usr/local/bin/:
------
failed to compute cache key: "/target/docker-configomat/configomat.sh" not found: not found

Do you know where it comes?
Regards

**Reply by casperklein** [reply]
There is a submodule missing. Use this to clone the repo: `git clone --recurse-submodules ..`

**Reply by jularnaud** [reply]
Thanks for your reply,

I actually solved the configomat.sh error with this command : git submodule update --init --recursive

Nevertheless, i still have the same version of dovecot (2.3.13) after creating a new container from this new image.
I am actually using docker-compose to run mailserver...

DOCKER_BUILDKIT=1 docker build --build-arg DOVECOT_COMMUNITY_REPO=1 -t mailserver .

Here is the compilation output :
[+] Building 286.9s (50/50) FINISHED                                                                                                                                                                                                                                          
 => [internal] load build definition from Dockerfile                                                                                                                                                                                                                     0.1s
 => => transferring dockerfile: 38B                                                                                                                                                                                                                                      0.0s
 => [internal] load .dockerignore                                                                                                                                                                                                                                        0.2s
 => => transferring context: 34B                                                                                                                                                                                                                                         0.0s
 => [internal] load metadata for docker.io/library/debian:11-slim                                                                                                                                                                                                        0.9s
 => [internal] load build context                                                                                                                                                                                                                                        0.2s
 => => transferring context: 7.74kB                                                                                                                                                                                                                                      0.0s
 => CACHED [ 1/46] FROM docker.io/library/debian:11-slim@sha256:xxx                                                                                                                                        0.0s
 => [ 2/46] RUN   apt-get -qq update &&   apt-get -qq install apt-utils 2>/dev/null &&   apt-get -qq dist-upgrade &&   echo "applying workaround for ubuntu/postfix bug described in https://github.com/docker-mailserver/docker-mailserver/issues/2023#issuecomment-  188.8s
 => [ 3/46] COPY ./target/scripts/helpers/log.sh /usr/local/bin/helpers/log.sh                                                                                                                                                                                           0.9s 
 => [ 4/46] COPY ./target/bin/sedfile /usr/local/bin/sedfile                                                                                                                                                                                                             0.7s 
 => [ 5/46] RUN chmod +x /usr/local/bin/sedfile                                                                                                                                                                                                                          1.3s 
 => [ 6/46] RUN   echo '0 */6 * * * clamav /usr/bin/freshclam --quiet' >/etc/cron.d/clamav-freshclam &&   chmod 644 /etc/clamav/freshclam.conf &&   freshclam &&   sedfile -i 's/Foreground false/Foreground true/g' /etc/clamav/clamd.conf &&   mkdir /var/run/clamav  36.5s 
 => [ 7/46] COPY target/dovecot/auth-passwdfile.inc target/dovecot/auth-master.inc target/dovecot/??-*.conf /etc/dovecot/conf.d/                                                                                                                                         1.3s 
 => [ 8/46] COPY target/dovecot/sieve/ /etc/dovecot/sieve/                                                                                                                                                                                                               0.7s 
 => [ 9/46] COPY target/dovecot/dovecot-purge.cron /etc/cron.d/dovecot-purge.disabled                                                                                                                                                                                    0.7s 
 => [10/46] RUN chmod 0 /etc/cron.d/dovecot-purge.disabled                                                                                                                                                                                                               1.3s 
 => [11/46] WORKDIR /usr/share/dovecot                                                                                                                                                                                                                                   0.8s 
 => [12/46] RUN   sedfile -i -e 's/include_try \/usr\/share\/dovecot\/protocols\.d/include_try \/etc\/dovecot\/protocols\.d/g' /etc/dovecot/dovecot.conf &&   sedfile -i -e 's/#mail_plugins = \$mail_plugins/mail_plugins = \$mail_plugins sieve/g' /etc/dovecot/conf.  1.4s 
 => [13/46] COPY target/dovecot/dovecot-ldap.conf.ext /etc/dovecot                                                                                                                                                                                                       0.7s
 => [14/46] COPY   target/postfix/ldap-users.cf   target/postfix/ldap-groups.cf   target/postfix/ldap-aliases.cf   target/postfix/ldap-domains.cf   target/postfix/ldap-senders.cf   /etc/postfix/                                                                       1.5s
 => [15/46] RUN   sedfile -i -r 's/^(CRON)=0/\1=1/g' /etc/default/spamassassin &&   sedfile -i -r 's/^\$INIT restart/supervisorctl restart amavis/g'     /etc/spamassassin/sa-update-hooks.d/amavisd-new &&   mkdir -p /etc/spamassassin/kam/ &&   curl -sSfLo /etc/spa  1.7s
 => [16/46] COPY target/postsrsd/postsrsd /etc/default/postsrsd                                                                                                                                                                                                          0.7s
 => [17/46] COPY target/postgrey/postgrey /etc/default/postgrey                                                                                                                                                                                                          0.6s
 => [18/46] COPY target/postgrey/postgrey.init /etc/init.d/postgrey                                                                                                                                                                                                      0.7s
 => [19/46] RUN   chmod 755 /etc/init.d/postgrey &&   mkdir /var/run/postgrey &&   chown postgrey:postgrey /var/run/postgrey &&   curl -Lsfo /etc/postgrey/whitelist_clients https://postgrey.schweikert.ch/pub/postgrey_whitelist_clients                               1.3s
 => [20/46] COPY target/amavis/conf.d/* /etc/amavis/conf.d/                                                                                                                                                                                                              0.8s
 => [21/46] RUN   sedfile -i -r 's/#(@|   \\%)bypass/\1bypass/g' /etc/amavis/conf.d/15-content_filter_mode &&   adduser clamav amavis &&   adduser amavis clamav &&   adduser --system syslog &&   useradd -u 5000 -d /home/docker -s /bin/bash -p "$(echo docker | ope  2.7s
 => [22/46] RUN   for _ in {1..10}; do su - amavis -c "razor-admin -create" ; sleep 3 ;   if su - amavis -c "razor-admin -register" ; then { EC=0 ; break ; } ;   else EC=${?} ; fi ; done ; (exit ${EC})                                                                7.2s
 => [23/46] COPY target/fail2ban/jail.local /etc/fail2ban/jail.local                                                                                                                                                                                                     0.8s 
 => [24/46] RUN   ln -s /var/log/mail/mail.log /var/log/mail.log &&   rm /etc/fail2ban/jail.d/defaults-debian.conf &&   mkdir /var/run/fail2ban                                                                                                                          1.1s 
 => [25/46] COPY target/opendkim/opendkim.conf /etc/opendkim.conf                                                                                                                                                                                                        0.8s 
 => [26/46] COPY target/opendkim/default-opendkim /etc/default/opendkim                                                                                                                                                                                                  0.7s 
 => [27/46] COPY target/opendmarc/opendmarc.conf /etc/opendmarc.conf                                                                                                                                                                                                     0.6s 
 => [28/46] COPY target/opendmarc/default-opendmarc /etc/default/opendmarc                                                                                                                                                                                               0.6s
 => [29/46] COPY target/opendmarc/ignore.hosts /etc/opendmarc/ignore.hosts                                                                                                                                                                                               0.6s
 => [30/46] RUN echo 'Reason_Message = Message {rejectdefer} due to: {spf}.' >>/etc/postfix-policyd-spf-python/policyd-spf.conf                                                                                                                                          1.3s
 => [31/46] COPY target/fetchmail/fetchmailrc /etc/fetchmailrc_general                                                                                                                                                                                                   0.8s
 => [32/46] COPY target/postfix/main.cf target/postfix/master.cf /etc/postfix/                                                                                                                                                                                           0.9s
 => [33/46] COPY target/shared/ffdhe4096.pem /etc/postfix/dhparams.pem                                                                                                                                                                                                   0.7s
 => [34/46] COPY target/shared/ffdhe4096.pem /etc/dovecot/dh.pem                                                                                                                                                                                                         0.7s
 => [35/46] COPY   target/postfix/header_checks.pcre   target/postfix/sender_header_filter.pcre   target/postfix/sender_login_maps.pcre   /etc/postfix/maps/                                                                                                             1.0s
 => [36/46] RUN   : >/etc/aliases &&   sedfile -i 's/START_DAEMON=no/START_DAEMON=yes/g' /etc/default/fetchmail &&   mkdir /var/run/fetchmail && chown fetchmail /var/run/fetchmail                                                                                      1.2s
 => [37/46] RUN   sedfile -i -r "/^#?compress/c\compress\ncopytruncate" /etc/logrotate.conf &&   mkdir -p /var/log/mail &&   chown syslog:root /var/log/mail &&   touch /var/log/mail/clamav.log &&   chown -R clamav:root /var/log/mail/clamav.log &&   touch /var/log  1.8s
 => [38/46] COPY target/logwatch/maillog.conf /etc/logwatch/conf/logfiles/maillog.conf                                                                                                                                                                                   0.8s
 => [39/46] COPY target/supervisor/supervisord.conf /etc/supervisor/supervisord.conf                                                                                                                                                                                     0.6s
 => [40/46] COPY target/supervisor/conf.d/* /etc/supervisor/conf.d/                                                                                                                                                                                                      0.7s
 => [41/46] RUN   rm -rf /usr/share/locale/* &&   rm -rf /usr/share/man/* &&   rm -rf /usr/share/doc/* &&   touch /var/log/auth.log &&   update-locale &&   rm /etc/postsrsd.secret &&   rm /etc/cron.daily/00logwatch                                                   1.4s
 => [42/46] COPY ./VERSION /                                                                                                                                                                                                                                             0.7s
 => [43/46] COPY   ./target/bin/*   ./target/scripts/*.sh   ./target/scripts/startup/*.sh   ./target/scripts/wrapper/*.sh   ./target/docker-configomat/configomat.sh   /usr/local/bin/                                                                                   1.6s
 => [44/46] RUN chmod +x /usr/local/bin/*                                                                                                                                                                                                                                2.2s
 => [45/46] COPY ./target/scripts/helpers /usr/local/bin/helpers                                                                                                                                                                                                         0.8s
 => exporting to image                                                                                                                                                                                                                                                   9.7s
 => => exporting layers                                                                                                                                                                                                                                                  9.7s
 => => writing image sha256:xxx                                                                                                                                                                            0.0s
 => => naming to docker.io/library/mailserver

how can we know if davecot has been installed from their repository?

Thanks for your reply,
